### PR TITLE
Metafeeds getsubset

### PR DESCRIPTION
**Pull request by cryptix**
This implements https://github.com/ssb-ngi-pointer/ssb-subset-replication-spec#getsubsetquery-options-source

- [x] Go query api
- [x] query planer
- [x] muxrpc handler 
- [x] add more options to rpc method (`pageLimit:n`, `reverse:bool`, `keys:bool`)

(based on #104 - merge first)

### CODE REVIEW

**Review by cryptix — COMMENTED** [review]
small review comment

File: query/subsetquery_test.go [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+type tcaseSerialze struct {
+	name string
+
+	jsonInput string
+	query     query.SubsetOperation
+
+	invalid bool
+}
+
+func (tc tcaseSerialze) run(t *testing.T) {
```
Comment (cblgh) [inline query/subsetquery_test.go:108]: `tcaseSerialze`should probably be `tcaseSerialize[d]`

File: query/subsetquery.go [hunk 2]
```diff
@@ -0,0 +1,105 @@
+// SPDX-License-Identifier: MIT
```
Comment (cblgh) [inline query/subsetquery.go:1]: very nicely commented :)

File: query/subsetquery.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	err := json.Unmarshal(input, &m)
+	if err != nil {
+		return fmt.Errorf("subset query unmarshaling failed: %w", err)
+	}
+
+	switch m.Operation {
+	case "and":
+		so.args = m.Args
+	case "or":
+		so.args = m.Args
```
Comment (cblgh) [inline query/subsetquery.go:78]:
```suggestion
	case "or", "and":
		so.args = m.Args

File: plugins/partial/subset.go [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+	// iterate over the combined set of bitmaps
+	var (
+		it *sroar.Iterator
+
+		buf bytes.Buffer
+		enc = json.NewEncoder(&buf)
+	)
+
+	if opts.Descending {
+		it = resulting.NewReverseIterator()
```
Comment (cblgh) [inline plugins/partial/subset.go:74]: formatting bugged out here?

**Review by cblgh — APPROVED** [review]
el gt em

**Pull request merged**